### DISCUSSION TITLE
2 color bars in a scale

### DISCUSSION THREAD

**Discussion by trudator** [mtex-toolbox/mtex#1638] [Ask Anything] [answered]
Hi,

What is the best way to put 2 color bars in one scale? Let's say from 0-1 is a black to white and 1-4 is a blue  to red one. 
![image](https://user-images.githubusercontent.com/117386810/200976521-92c21a53-e3db-4ce6-b182-7337df4b7174.png)

Thanks.

**Comment by kilir** [comment]
Hi,
if it is a colorbar which is always supposed to be greyvalues from 0-1 and 1-4, you can simply write an n by 3 matrix containing the appropriate values in the range [0,1]. (Grey is always the same entries for RGB, each row). If he maximum you want to display changes, you'd need to adapt it.
Cheers,
Rüdiger

**Reply by trudator** [reply]
Hi Rudiger,

Thanks for your answer but not very clear to me, sorry I'm a newbie here. Could you clarify how the matrix is structured and why we need an nx3 matrix for 0-1 and 1-4? What if I change the latter to 1-8?

**Comment by kilir** [comment, marked as answer]
For example:
```
cm = [gray(40); parula(120)]

mtexdata titanium

somevalue_from_0to4 = (ebsd.iq - min(ebsd.iq)) * ...
                      4/max(ebsd.iq - min(ebsd.iq))

plot(ebsd,somevalue_from_0to4)
mtexColorMap(cm)
```

![example](https://user-images.githubusercontent.com/11697557/201188943-9f056069-2f16-41a8-ab08-035bd6250134.png)

**Reply by trudator** [reply]
So the higher magnitude in the parenthesis the higher resolution of our colormap?
By the way, it works perfectly. Thanks, Rudiger!

**Reply by kilir** [reply]
Hi,
great that it worksfor you. Nothing about resolutionbut just stacking together two colormaps at the right proportions.
Cheers,
Rüdiger

**Reply by trudator** [reply]
Thanks
